tests/flow/tests_withGears.py

```diff
@@ -0,0 +1,326 @@
+    except Exception as e:
+        return e
+        
+async def DAGRun_addOpsFromString(record):
+    
+    keys = ['a{1}', 'b{1}']
+    tensors = redisAI.mgetTensorsFromKeyspace(keys)
+    DAGRunner = redisAI.createDAGRunner()
+    DAGRunner.Input('tensor_a', tensors[0]).Input('tensor_b', tensors[1])
+    DAGRunner.OpsFromString('|> AI.MODELRUN m{1} INPUTS tensor_a tensor_b OUTPUTS tensor_c |> AI.TENSORGET tensor_c')
+    res = await DAGRunner.Run()
+    redisAI.setTensorInKey('test5_res{1}', res[0])
+    return "test5_OK"
+        
+GB("CommandReader").map(DAGRun_tensorSetTensorGet).register(trigger="DAGRun_test1")
+GB("CommandReader").map(DAGRun_simpleModelRun).register(trigger="DAGRun_test2")
+GB("CommandReader").map(DAGRun_simpleScriptRun).register(trigger="DAGRun_test3")
+GB("CommandReader").map(DAGRun_scriptRunError).register(trigger="DAGRun_test4")
+GB("CommandReader").map(DAGRun_addOpsFromString).register(trigger="DAGRun_test5")
+    '''
+
+    con = env.getConnection()
+    ret = con.execute_command('rg.pyexecute', script)
+    env.assertEqual(ret, b'OK')
+
+    ret = con.execute_command('AI.TENSORSET', 'a{1}', 'FLOAT', 2, 2, 'VALUES', 2, 3, 2, 3)
+    env.assertEqual(ret, b'OK')
+    ret = con.execute_command('rg.trigger', 'DAGRun_test1')
+    env.assertEqual(ret[0], b'test1_OK')
+
+    values = con.execute_command('AI.TENSORGET', 'test1_res{1}', 'VALUES')
+    env.assertEqual(values, [b'2', b'3', b'2', b'3'])
+
+    con.execute_command('AI.TENSORSET', 'b{1}', 'FLOAT',
+                        2, 2, 'VALUES', 2, 3, 2, 3)
+    test_data_path = os.path.join(os.path.dirname(__file__), 'test_data')
+    model_filename = os.path.join(test_data_path, 'graph.pb')
+
+    with open(model_filename, 'rb') as f:
+        model_pb = f.read()
+    ret = con.execute_command('AI.MODELSET', 'm{1}', 'TF', DEVICE,
+                              'INPUTS', 'a', 'b', 'OUTPUTS', 'mul', 'BLOB', model_pb)
+    env.assertEqual(ret, b'OK')
+    ret = con.execute_command('rg.trigger', 'DAGRun_test2')
+    env.assertEqual(ret[0], b'test2_OK')
+
+    values = con.execute_command('AI.TENSORGET', 'test2_res{1}', 'VALUES')
+    env.assertEqual(values, [b'4', b'9', b'4', b'9'])
+
+    script_filename = os.path.join(test_data_path, 'script.txt')
+    with open(script_filename, 'rb') as f:
+        script = f.read()
+    ret = con.execute_command('AI.SCRIPTSET', 'myscript{1}', DEVICE, 'TAG', 'version1', 'SOURCE', script)
+    env.assertEqual(ret, b'OK')
+
+    ret = con.execute_command('rg.trigger', 'DAGRun_test3')
+    env.assertEqual(ret[0], b'test3_OK')
+
+    values = con.execute_command('AI.TENSORGET', 'test3_res{1}', 'VALUES')
+    env.assertEqual(values, [b'4', b'6', b'4', b'6'])
+
+    ret = con.execute_command('rg.trigger', 'DAGRun_test4')
+    # This should raise an exception
+
+    error_string = b'attempted to get undefined function no_func'
+    env.assertEqual(str(ret[0])[:len(error_string)+2]+"'", "{}".format(error_string))
```

Comment:
what is this?

Reply:
An assertion that the error message is the one we expect to. Actually the error message is much longer and detailed, so we only verify that it starts with error_string. I changed line 319 so that the code will be more understandable (but it does the same)